### ISSUE TITLE
RemoteEvaluationClient not properly cleaning up connections when timeout occurs

### ISSUE THREAD

**Issue by nickegg** [amplitude/experiment-python-server#22]
## Expected Behavior
The Remote Evaluation client should properly close the connection within the connection pool when a timeout occurs.

## Current Behavior
The connection in the connection pool is not being closed and is left in a bad state which causes subsequent calls to fail with request-sent

```python
[Experiment] Retry failed: Request-sent
[ERROR] (Amplitude) [Experiment] Retry failed: Request-sent
```

## Possible Solution
Handle timeout exceptions by setting `is_available` to `False` in order to properly clean up connections.
```python
try:
    response = conn.request('POST', '/sdk/vardata', body, headers)
    elapsed = '%.3f' % ((time.time() - start) * 1000)
    self.logger.debug(f"[Experiment] Fetch complete in {elapsed} ms")
    json_response = json.loads(response.read().decode("utf8"))
    variants = self.__parse_json_variants(json_response)
    self.logger.debug(f"[Experiment] Fetched variants: {json.dumps(variants, default=str)}")
    return variants
except TimeoutError as e:
    conn.is_available = False
    raise e
finally:
    self._connection_pool.release(conn)
```

## Steps to Reproduce
 

1.  Set the timeout to a very low value 
```python
evaluation_config = RemoteEvaluationConfig(fetch_retries=1, fetch_timeout_millis=1)
Experiment.initialize_remote(api_key, evaluation_config)
```
3. Call Amplitude 
```python
experiment.fetch_async(user, callback=callback)
```
4. Observe the connection is not disposed and as such all calls after fail
```python
[Experiment] Retry failed: Request-sent
[ERROR] (Amplitude) [Experiment] Retry failed: Request-sent
```

## Environment
- SDK Version: 1.1.1
- Language Version: 3.11.2

**Comment by bgiori** [comment]
Hi @nickegg, thanks for reporting this issue. I should be able to get to this very soon.

**Comment by bgiori** [comment]
@nickegg this should be fixed in `v1.1.2`. Thanks again for submitting the thorough issue, and please let me know if theres anything else.
